# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/lld/hotfix-gnu-prop-section.patch
# Copyright (C) 2023 - 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- lld-20.1.0.src/ELF/InputFiles.cpp.orig	2025-03-04 20:51:43.000000000 +0100
+++ lld-20.1.0.src/ELF/InputFiles.cpp	2025-03-05 16:31:12.889486269 +0100
@@ -939,13 +939,12 @@
   while (!data.empty()) {
     // Read one NOTE record.
     auto *nhdr = reinterpret_cast<const Elf_Nhdr *>(data.data());
-    if (data.size() < sizeof(Elf_Nhdr) ||
-        data.size() < nhdr->getSize(sec.addralign))
+    if (data.size() < sizeof(Elf_Nhdr) || data.size() < nhdr->getSize(1))
       return void(err(data.data()) << "data is too short");
 
     Elf_Note note(*nhdr);
     if (nhdr->n_type != NT_GNU_PROPERTY_TYPE_0 || note.getName() != "GNU") {
-      data = data.slice(nhdr->getSize(sec.addralign));
+      data = data.slice(nhdr->getSize(1));
       continue;
     }
 
@@ -954,7 +953,7 @@
                                   : GNU_PROPERTY_X86_FEATURE_1_AND;
 
     // Read a body of a NOTE record, which consists of type-length-value fields.
-    ArrayRef<uint8_t> desc = note.getDesc(sec.addralign);
+    ArrayRef<uint8_t> desc = note.getDesc(1);
     while (!desc.empty()) {
       const uint8_t *place = desc.data();
       if (desc.size() < 8)
@@ -993,7 +992,7 @@
     }
 
     // Go to next NOTE record to look for more FEATURE_1_AND descriptions.
-    data = data.slice(nhdr->getSize(sec.addralign));
+    data = data.slice(nhdr->getSize(1));
   }
 }
 
